modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/jdbc/JdbcMetadataCatalog.java

```diff
@@ -182,28 +202,24 @@ public CompletableFuture<Collection<String>> getSchemasMeta(String schemaNamePtr
             schemas.add(DEFAULT_SCHEMA_NAME);
         }
 
-        return tables.tablesAsync().thenApply(tablesList ->
-                tablesList.stream()
+        return tablesAtNow().thenApply(tables ->
+                tables.stream()
                     .map(tbl -> DEFAULT_SCHEMA_NAME)
                     .filter(schema -> matches(schema, schemaNameRegex))
-                    .collect(Collectors.toCollection(() -> schemas))
+                    .collect(toCollection(() -> schemas))
         );
     }
 
     /**
-     * Creates primary key metadata from table object.
+     * Creates primary key metadata from a table descriptor.
      *
      * @param tbl Table.
      * @return Jdbc primary key metadata.
      */
-    private JdbcPrimaryKeyMeta createPrimaryKeyMeta(Table tbl) {
-        final String keyName = PK + tbl.name();
-
-        SchemaRegistry registry = ((TableImpl) tbl).schemaView();
+    private JdbcPrimaryKeyMeta createPrimaryKeyMeta(CatalogTableDescriptor tbl) {
+        String keyName = PK + tbl.name();
 
-        List<String> keyColNames = Arrays.stream(registry.schema().keyColumns().columns())
-                .map(Column::name)
-                .collect(Collectors.toList());
+        List<String> keyColNames = new ArrayList<>(tbl.primaryKeyColumns());
```

Comment:
Any particular reason of creating the new list?

Reply:
To make sure that, if `tbl.primaryKeyColumns()` is modifiable, we don't give it away

Reply:
Replaced it with `List.copyOf()` which will not allocate anything if the original list is immutable

Reply:
I assume that if the list is returned from a method to external user, we can be user that it can be used for reading. Such issues are solved by documentation. Making copies of data, just to be sure, might not be the best solution

Reply:
I disagree. Noone reads documentation. The pattern is called 'defensive copy', it's widely used. Also, please note that this code is not on the hot path. A couple of instances on the heap will not crash the server.

Reply:
> The pattern is called 'defensive copy', it's widely used.

You know that's bad for performance, right? How hot is this code?